bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/model/local/X3HybridG4Data.java

```diff
@@ -168,47 +169,47 @@ public short getPowerUsage() {
 
     @Override
     public double getTotalEnergy() {
-        return ((double) getFromRawData(68)) / 10;
+        return ((double) ByteUtil.read32BitSigned(getFromRawData(68), getFromRawData(69))) / 10;
     }
 
     @Override
-    public short getTotalBatteryDischargeEnergy() {
-        return getFromRawData(74);
+    public double getTotalBatteryDischargeEnergy() {
+        return ((double) ByteUtil.read32BitSigned(getFromRawData(74), getFromRawData(75))) / 10;
     }
 
     @Override
-    public short getTotalBatteryChargeEnergy() {
-        return getFromRawData(76);
+    public double getTotalBatteryChargeEnergy() {
+        return ((double) ByteUtil.read32BitSigned(getFromRawData(76), getFromRawData(77))) / 10;
     }
 
     @Override
     public double getTotalPVEnergy() {
-        return ((double) getFromRawData(80)) / 10;
+        return ((double) ByteUtil.read32BitSigned(getFromRawData(80), getFromRawData(81))) / 10;
     }
 
     @Override
-    public short getTotalFeedInEnergy() {
-        return getFromRawData(86);
+    public double getTotalFeedInEnergy() {
+        return ((double) ByteUtil.read32BitSigned(getFromRawData(86), getFromRawData(87))) / 100;
     }
 
     @Override
     public double getTotalConsumption() {
-        return ((double) getFromRawData(88)) / 10;
+        return ((double) ByteUtil.read32BitSigned(getFromRawData(88), getFromRawData(89))) / 10;
     }
 
     @Override
     public double getTodayEnergy() {
-        return ((double) getFromRawData(82)) / 10;
+        return ((double) getFromRawData(70)) / 10;
```

Comment:
Seems like a complete different position, just a double check this is not a mistake.

Reply:
Not a mistake. Confirmed by the community to be the right index for these inverters. Tests adapted to the new values.